.github/workflows/main.yml

```diff
@@ -102,20 +141,57 @@ jobs:
       # ensure that the tests fail more quickly so that people can fix them.
       timeout-minutes: 30
 
-    - name: Build executor image
-      run: ./tools/executor/build.js
+  native-checks:
+    runs-on: ubuntu-latest
 
-    # The following steps won't run correctly for forks since the Dockerhub
-    # credentials secrets will be missing. We only run them if the secrets are
-    # provided. Note that we can't use `secrets.*` directly in an `if` conditional
-    # here, so we copy them to an environment variable at the start of the job
-    # and then check that here.
-    - name: Login to DockerHub
-      uses: docker/login-action@v2
-      if: ${{ env.DOCKERHUB_USERNAME_SECRET != '' }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install OS packages
+      run: sudo apt-get install -y graphviz graphviz-dev
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
       with:
-        username: ${{ secrets.DOCKERHUB_USERNAME }}
-        password: ${{ secrets.DOCKERHUB_TOKEN }}
-    - name: Push executor image to Docker registry
-      if: ${{ env.DOCKERHUB_USERNAME_SECRET != '' }}
-      run: ./tools/executor/release.js
+        python-version: '3.10'
+        cache: pip
+        cache-dependency-path: images/plbase/python-requirements.txt
+    - name: Install Python dependencies
+      run: pip install -r images/plbase/python-requirements.txt
+
+    - name: Set up Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16'
+        cache: 'yarn'
+    - name: Install Node dependencies
+      run: yarn install --immutable
+
+    - name: Set up Turborepo cache
+      uses: actions/cache@v3
+      with:
+        path: ./node_modules/.cache/turbo
+        key: ${{ runner.os }}-turbo-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-turbo-
```

Comment:
This allows us to cache the results of `make build` (which is `turbo run build` under the hood).